seq2simulate/run_make_proviral_data.py

```diff
@@ -0,0 +1,447 @@
+        sequences_strings, seq_diffs = mutation_func(sequences_strings, **mutation_kwargs)
+        seq_diffs = dict(zip(sequence_ids, seq_diffs))
+        diversity._update_seq_reads_from_strs(sequences, sequences_strings)
+        data_files[mutation_type] = (_write_to_FASTA(sequences, working_dir, '{}_data.fasta'.format(i_mutation)))
+        diff_files[mutation_type] = (_write_to_file(seq_diffs, working_dir, '{}_diffs.data'.format(i_mutation)))
+
+    return data_files, diff_files
+
+
+def _make_art_files(data_files, platform, working_dir, paired_end=False):
+    platf = getattr(plat, platform)
+
+    fastq_files, sam_files = {}, {}
+    for mutation_type, data_file in data_files.iteritems():
+        fastq_file, sam_file = art.simulate(data_file, platf, platf.coverage, paired_end, working_dir)
+        if not isinstance(fastq_file, tuple):
+            fastq_file = [fastq_file]
+        fastq_files[mutation_type], sam_files[mutation_type] = [i for i in fastq_file], sam_file
+    return fastq_files, sam_files
+
+
+def _open_all_data_files(sequences, fastq_files, sam_files, diff_files):
+    open_fq_files = {}
+    for mutation_type, fq in fastq_files.iteritems():
+        fastq_siblings = []
+        for _fq in fq:
+            with open(_fq, 'r') as f:
+                fastq_siblings.append([i for i in parse_fastq(f)])
+        open_fq_files[mutation_type] = fastq_siblings
+    
+    open_sam_files = {}
+    for mutation_type, fs in sam_files.iteritems():
+        with open(fs, 'r') as f:
+            sam_reads = {}
+            for sam_read in parse_sam(f):
+                read_id = sam_read[0].strip('\n') 
+                if read_id in sam_reads:
+                    sam_reads[read_id].append(sam_read[1:])
+                else:
+                    sam_reads[read_id] = [sam_read[1:]]
+            open_sam_files[mutation_type] = sam_reads
+
+    open_diff_files = {'null': {i.id: [] for i in sequences}}
+    for mutation_type, fd in diff_files.iteritems():
+        with open(fd, 'r') as f:
+            open_diff_files[mutation_type] = pickle.load(f)
+    return open_fq_files, open_sam_files, open_diff_files
+
+
+def _decorate_fastq_headers(fastq_sample, mutation_type, sam_file=None, diff_file=None, paired_end=False):
+
+    """
+    Decorate FASTQ headers with a suffix code indicating whether or not this
+    read covers a known mutation (see MUTATIONS for mutation codes).
+    """
+    for i_read, read in enumerate(fastq_sample):
+        id_suffix = None
+        if mutation_type != 'null':
+            if sam_file is not None and diff_file is not None:
+                read_id = read[0][1:].strip('\n')
+                if paired_end:
+                    read_id, read_direction = read_id[:-2], read_id[-1]
+                    sam_line_f, sam_line_r = _parse_sam_line(read_id, sam_file, paired_end=paired_end)
+                    if read_direction == '1':
```

Comment:
/1 and /2 don't actually imply directionality (forward/reverse), just "first" and "second".  You need to get the actual direction from the sam file (the 0x10 bit of the FLAG will be set for reverse).